jenkins/test/run_tests.py

```diff
@@ -0,0 +1,200 @@
+    "common.py",
+    "yaml_validation.py",
+    ".pylintrc"
+]
+VALIDATOR_PATH = "jenkins/test/validators/"
+GITHUB_API_URL = "https://api.github.com"
+REPO = "openshift-tools"
+REPO_USER = "openshift"
+
+def run_cli_cmd(cmd, exit_on_fail=True):
+    '''Run a command and return its output'''
+    proc = subprocess.Popen(cmd, bufsize=-1, stderr=subprocess.PIPE, stdout=subprocess.PIPE,
+                            shell=False)
+    stdout, stderr = proc.communicate()
+    if proc.returncode != 0:
+        print "Unable to run " + " ".join(cmd) + " due to error: " + stderr
+        if exit_on_fail:
+            sys.exit(proc.returncode)
+        else:
+            return False, stdout
+    else:
+        return True, stdout
+
+def assign_env(pull_request):
+    '''Assign environment variables base don github webhook payload json data'''
+    # Github environment variables
+    os.environ["PRV_TITLE"] = pull_request["title"]
+    os.environ["PRV_BODY"] = pull_request["body"]
+    os.environ["PRV_PULL_ID"] = pull_request["number"]
+    os.environ["PRV_URL"] = pull_request["url"]
+
+    # Base environment variables
+    base = pull_request["base"]
+    os.environ["PRV_BASE_SHA"] = base["sha"]
+    os.environ["PRV_BASE_REF"] = base["ref"]
+    os.environ["PRV_BASE_LABEL"] = base["label"]
+    os.environ["PRV_BASE_NAME"] = base["repo"]["full_name"]
+
+    # Remote environment variables
+    head = pull_request["head"]
+    os.environ["PRV_REMOTE_SHA"] = head["sha"]
+    os.environ["PRV_REMOTE_REF"] = head["ref"]
+    os.environ["PRV_REMOTE_LABEL"] = head["label"]
+    os.environ["PRV_REMOTE_NAME"] = head["repo"]["full_name"]
+
+def merge_changes(pull_request):
+    """ Merge changes into current repository """
+    remote_url = pull_request["head"]["repo"]["git_url"]
+    remote_ref = pull_request["head"]["ref"]
+    base_ref = pull_request["base"]["ref"]
+    test_branch_name = "tpr-" + remote_ref + "-" + pull_request["number"]
+
+    run_cli_cmd(['/usr/bin/git', 'checkout', base_ref])
+    run_cli_cmd(['/usr/bin/git', 'checkout', '-b', test_branch_name])
+    run_cli_cmd(['/usr/bin/git', 'remote', 'add', 'remoterepo', remote_url])
+    run_cli_cmd(['/usr/bin/git', 'fetch', 'remoterepo', remote_ref])
+
+def run_validators():
+    """ Run all test validators """
+    failure_occured = False
+    validators = [validator for validator in os.listdir(VALIDATOR_PATH) if
+                  os.path.isfile(os.path.join(VALIDATOR_PATH, validator))]
+    for validator in validators:
+        skip = False
+        for exclude in EXCLUDES:
+            if validator == exclude:
+                skip = True
+        if skip:
+            continue
+        validator_abs = os.path.join(VALIDATOR_PATH, validator)
+        executer = ""
+        _, ext = os.path.splitext(validator)
+        if ext == ".py":
+            executer = "/usr/bin/python"
+        elif ext == ".sh":
+            executer = "/bin/sh"
+	# If the ext is not recongized, try to just run the file
```

Comment:
This makes me nervous and we should chat about it.  I won't block for now but I can see this being a black hole.  I believe this would allow someone to run code in our jenkins instance, is that the case?